app/src/main/scala/org/alephium/explorer/service/BlockFlowClient.scala

```diff
@@ -323,9 +323,14 @@ object BlockFlowClient {
     )
 
   private def inputToUInput(input: api.model.AssetInput): UInput = {
+    val unlockScript = input.toProtocol() match {
+      case Left(error)  => throw new ExplorerError.InvalidProtocolInput(error)
+      case Right(value) => value.unlockScript
+    }
```

Comment:
agree on the function to remove the duplicate.

For the `Try`, we would need to propagate the `Try` up to the top and we endup with some `Seq[Try[A]]`, it seems to bring a lot of complexity while throwing an error is also failing the Future right?

Reply:
Yep you are right. `throw` will also fail the `Future`. Just a syntax thing because if I have another function that is not within the `Future`, I will also need to learn what's in this function so that new function can handle the `throw`. If it's returning `Try`, I wouldn't need to learn about this function.

Reply:
oh good point, I could indeed return a `Try`, but throw from where I use that function